### PR TITLE
Add Docker configuration so the service can be run in a Docker container

### PR DESCRIPTION
**Pull request by JoeChapman**
- add Dockerfile
- add docker-compose.yml
- add run.sh

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+if [ "$NODE_ENV" = "ci" ]
+then echo "starting service"
+  SITEROOT=/firearms GA_TAG_ID=UA-72527484-1
```
Comment (easternbloc) [inline run.sh]: is that the correct GA TAG?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+version: "2"
```
Comment (bendiggle) [inline docker-compose.yml]: Should it be v2?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+services:
```
Comment (bendiggle) [inline docker-compose.yml]: Also, is firearms not going to need maildev?

Comment (easternbloc) [inline docker-compose.yml]: Probably not. At least not right now.

File: docker-compose.yml (hunk 2 above)
Comment (easternbloc) [inline docker-compose.yml]: Yes, We've migrated to compose v2

File: run.sh (hunk 1 above)
Comment (JoeChapman) [inline run.sh]: TBH, I am not sure what SITEROOT is for. Does anyone lnow if it's still required? @easternbloc @gxxm @joefitter @bendiggle ???

Comment (easternbloc) [inline run.sh]: it's used to make assetPath which is then uses across our templates.

Comment (easternbloc) [inline run.sh]: Not sure why it's prefixed with the app name though. BRP doesn't do this...

Comment (easternbloc) [inline run.sh]: Sadly bad commit messages do not explain it: https://github.com/UKHomeOffice/gro/commit/069eca1f

Comment (JoeChapman) [inline run.sh]: Looks like BS to me. Why would baseUrl ever need to be `firearms/new-dealer` or `brp/not-arrived` or such like? It's as if the app is serving from a level above root, it's wrong.

**Comment by JoeChapman** [comment]
@easternbloc @bendiggle?

File: run.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+if [ "$NODE_ENV" = "ci" ]
+then echo "starting service"
+  GA_TAG_ID=
```
Comment (easternbloc) [inline run.sh]: have you created a ticket for analytics so this doesn't get missed...

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      - redis
+    ports:
+      - "8080:8080"
+    volumes:
+      - "/public"
+  redis:
+    image: redis
+
+  nginx-proxy:
+    image: quay.io/ukhomeofficedigital/nginx-proxy:v1.2.3
```
Comment (easternbloc) [inline docker-compose.yml]: the latest nginx-proxy is 1.5.1 I'd suggest updating to that as it's a fresh project

File: run.sh (hunk 4 above)
Comment (JoeChapman) [inline run.sh]: Yes: https://jira.digital.homeoffice.gov.uk/browse/FLHO-93

**Comment by bendiggle** [comment]
LGTM

**Pull request merged**